src/main/java/de/dennisguse/opentracks/viewmodels/IntervalStatistics.java

```diff
@@ -0,0 +1,80 @@
+package de.dennisguse.opentracks.viewmodels;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.dennisguse.opentracks.content.data.TrackPoint;
+import de.dennisguse.opentracks.util.LocationUtils;
+import de.dennisguse.opentracks.util.UnitConversions;
+
+public class IntervalStatistics {
+    List<Interval> intervalList;
+
+    public void build(List<TrackPoint> trackPointList, float distanceInterval) {
+        intervalList = new ArrayList<>();
```

Comment:
Default initialize in l:11 and then use intervalList.clear();

This may save some memory allocation (or the runtime is really good).